superset/assets/javascripts/welcome/App.jsx

```diff
@@ -17,24 +23,65 @@ export default class App extends React.PureComponent {
   render() {
     return (
       <div className="container welcome">
-        <Panel>
-          <Row>
-            <Col md={8}><h2>Dashboards</h2></Col>
-            <Col md={4}>
-              <FormControl
-                type="text"
-                bsSize="sm"
-                style={{ marginTop: '25px' }}
-                placeholder="Search"
-                value={this.state.search}
-                onChange={this.onSearchChange}
-              />
-            </Col>
-          </Row>
-          <hr />
-          <DashboardTable search={this.state.search} />
-        </Panel>
+        <Tabs defaultActiveKey={1} id="uncontrolled-tab-example">
+          <Tab eventKey={1} title="Recently Viewed">
+            <Panel>
+              <Row>
+                <Col md={8}><h2>Recent Viewed</h2></Col>
+              </Row>
+              <hr />
+              <RecentActivity user={this.props.user} />
+            </Panel>
+          </Tab>
+          <Tab eventKey={2} title="Favorites">
+            <Panel>
+              <Row>
+                <Col md={8}><h2>Favorites</h2></Col>
+              </Row>
+              <hr />
+              <Favorites user={this.props.user} />
+            </Panel>
+          </Tab>
+          <Tab eventKey={3} title="Dashboards">
+            <Panel>
+              <Row>
+                <Col md={8}><h2>Dashboards</h2></Col>
+              </Row>
+              <hr />
+              <DashboardTable search={this.state.search} />
+            </Panel>
+          </Tab>
+          <Tab eventKey={4} title="Charts" disabled>
+            <Panel>
+              <Row>
+                <Col md={8}><h2>Datasources</h2></Col>
+              </Row>
+              <hr />
+              {/* <ChartsTable search={this.state.search} /> */}
```

Comment:
Why the empty placeholders? I'm thinking it's better to not have the empty tabs than having them.